Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -92,15 +95,27 @@ final class TransactionPoster: TransactionPosterType {
 
         if systemInfo.dangerousSettings.internalSettings.usesStoreKit2JWS,
            let jwsRepresentation = transaction.jwsRepresentation {
-            self.fetchProductsAndPostReceipt(
-                transaction: transaction,
-                data: data,
-                receipt: .jws(jwsRepresentation),
-                completion: completion
-            )
+            if transaction.environment == .xcode, #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+                _ = Task<Void, Never> {
```

Comment:
Any better way to avoid duplication? The enclosing function is not async

Reply:
Suggested below.